runner/entrypoint.sh

```diff
@@ -44,63 +64,85 @@ if [ -z "${RUNNER_REPO}" ] && [ -n "${RUNNER_GROUP}" ];then
 fi
 
 # Hack due to https://github.com/actions-runner-controller/actions-runner-controller/issues/252#issuecomment-758338483
-if [ ! -d /runner ]; then
-  echo "/runner should be an emptyDir mount. Please fix the pod spec." 1>&2
+if [ ! -d ${RUNNER_HOME} ]; then
+  error "${RUNNER_HOME} should be an emptyDir mount. Please fix the pod spec."
   exit 1
 fi
 
-sudo chown -R runner:docker /runner
-mv /runnertmp/* /runner/
-
-cd /runner
-./config.sh --unattended --replace \
-  --name "${RUNNER_NAME}" \
-  --url "${GITHUB_URL}${ATTACH}" \
-  --token "${RUNNER_TOKEN}" \
-  --runnergroup "${RUNNER_GROUPS}" \
-  --labels "${RUNNER_LABELS}" \
-  --work "${RUNNER_WORKDIR}"
-
-if [ -f /runner/.runner ]; then
-  echo Runner has successfully been configured with the following data.
-  cat /runner/.runner
-  # Note: the `.runner` file's content should be something like the below:
-  #
-  # $ cat /runner/.runner
-  # {
-  # "agentId": 117, #=> corresponds to the ID of the runner
-  # "agentName": "THE_RUNNER_POD_NAME",
-  # "poolId": 1,
-  # "poolName": "Default",
-  # "serverUrl": "https://pipelines.actions.githubusercontent.com/SOME_RANDOM_ID",
-  # "gitHubUrl": "https://github.com/USER/REPO",
-  # "workFolder": "/some/work/dir" #=> corresponds to Runner.Spec.WorkDir
-  # }
-  #
-  # Especially `agentId` is important, as other than listing all the runners in the repo,
-  # this is the only change we could get the exact runnner ID which can be useful for further
-  # GitHub API call like the below. Note that 171 is the agentId seen above.
-  #   curl \
-  #     -H "Accept: application/vnd.github.v3+json" \
-  #     -H "Authorization: bearer ${GITHUB_TOKEN}"
-  #     https://api.github.com/repos/USER/REPO/actions/runners/171
+# if this is not a testing environment
+if [ -z "$UNITTEST" ]; then
```

Comment:
We can't do this during unit testing, so had to add a conditional here ... not sure how to improve it but open to suggestions.